### DISCUSSION TITLE
怎么在umijs打包出来的umi.js和umi.css路径上添加前缀？

### DISCUSSION THREAD

**Discussion by xf11075** [umijs/umi#11881] [Q&A] [answered]
我们项目有个需求是要把资源托管到cdn，并支持cdn访问失败时回退到之前的url。

也就是生成的index.html中，umi打包进去的"./umi.js"和"./umi.css"需要修改为"https://cdn.cn/umi.js"，"https://cdn.cn/umi.css"
这个需求怎么实现呢？
我需要判断出所有umi打包添加的js和css路径，排除掉原本不需要修改的路径，也就是document.ejs中已有的js。

有没有什么api或钩子函数可以修改打包后输出的umi.js和umi.css路径的？

**Comment by fz6m** [comment, marked as answer]
产物的静态资源 `dist` 整个文件夹发到 CDN 上的话，要配 [`publicPath`](https://umijs.org/docs/api/config#publicpath) ，比如配了 `https://domain.com/path/` ，最后就会从 `https://domain.com/path/umi.js` 加载：

```ts
// .umirc.ts

  publicPath: process.env.NODE_ENV === 'production' ? 'https://domain.com/path/' : '/'
```

**Reply by xileliu** [reply]
如果是动态怎么生成

**Reply by xiaohuoni** [reply]
https://umijs.org/docs/api/config#runtimepublicpath
